crates/openvino/src/blob.rs

```diff
@@ -53,22 +53,25 @@ impl Blob {
     pub fn tensor_desc(&self) -> Result<TensorDesc> {
         let blob = self.instance as *const ie_blob_t;
 
-        let mut layout = Layout::ANY;
-        try_unsafe!(ie_blob_get_layout(blob, std::ptr::addr_of_mut!(layout)))?;
+        let mut layout = MaybeUninit::uninit();
+        try_unsafe!(ie_blob_get_layout(blob, layout.as_mut_ptr()))?;
 
-        let mut dimensions = dimensions_t {
-            ranks: 0,
-            dims: [0; 8usize],
-        };
-        try_unsafe!(ie_blob_get_dims(blob, std::ptr::addr_of_mut!(dimensions)))?;
+        let mut dimensions = MaybeUninit::uninit();
+        try_unsafe!(ie_blob_get_dims(blob, dimensions.as_mut_ptr()))?;
 
-        let mut precision = Precision::UNSPECIFIED;
-        try_unsafe!(ie_blob_get_precision(
-            blob,
-            std::ptr::addr_of_mut!(precision)
-        ))?;
+        let mut precision = MaybeUninit::uninit();
+        try_unsafe!(ie_blob_get_precision(blob, precision.as_mut_ptr()))?;
 
-        Ok(TensorDesc::new(layout, &dimensions.dims, precision))
+        Ok(TensorDesc {
+            // Safety: all reads succeeded so values must be initialized
+            instance: unsafe {
+                tensor_desc_t {
+                    layout: layout.assume_init(),
+                    dims: dimensions.assume_init(),
```

Comment:
I think I'm fine with the `MaybeUninit` changes but can we add a double-check assertion that `dimension.dims` is an array of size 8? `crates/openvino-sys/src/generated/types.rs` has a definition like the following:

```rust
pub struct dimensions {
    pub ranks: usize,
    pub dims: [usize; 8usize],
}
```

And if for some reason some version of OpenVINO changed this, I would want to know with a panic, not some undefined behavior. See `TensorDesc::new()` for how this looks there.

Reply:
I'm not sure I understand what you mean. `MaybeUninit` type is automatically inferred from `dimensions_t` (aka `dimensions`). So in this case it is `MaybeUninit<dimensions_t>` and if `dimensions` were changed, the type of `MaybeUninit` would change accordingly.

Or do you want to ensure that when bindings are regenerated `dimensions_t` dims remain a size of 8? In that case I believe a compile time assertion would be better than a runtime check.

Reply:
Unfortunately it is probably too complicated to do so in the bindings since they are auto-generated and that still doesn't solve the undefined-ness of potentially storing a wrongly-sized array in that field. It's a bit paranoid of me, I guess (why would OpenVINO itself return the wrong size anyways!?), but I would feel better if there is some check there. I'll merge this as-is and add something in a follow-on commit.